Sprint 6/step 1

anufrp opened the pull request.

anufrp committed:
reducers jest testing

Тестирование всех редьюсеров

anufrp committed:
Cypress testing

Тестирование Cypress согласно брифу

anufrp committed:
GH Pages deploy

Публикация проекта на страницу GitHub

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

К сожалению, работу придется отклонить без потери итерации

* Приложение ломается при перезагрузке страницы с открытым попапом https://disk.yandex.ru/i/Jl13ff7EMyRZNw 
* если **перезагрузить** страницу с открытым попапом заказа, то этот открытый попап опять должен появляться (или страница с деталями). А сейчас ничего не отображается после перезагрузки. Такая же проблема при вставке ссылки в браузер
* в **истории** не отображается заказ при клике на него https://disk.yandex.ru/i/I56gcWjO8lTjgA 
* очень много типов `any`, которые можно просто удалить. Скрин https://disk.yandex.ru/i/yp-CGFsl35s-vA  . Особенно в `useSelector((store: any)`

anufrp committed:
Работа над перезагрузкой модалки

anufrp committed:
Routing refactoring

Переделка ротинга в части информации о заказе на useHistory

anufrp commented:
Уточнил работу с веб-сокетом с помощью дополнительных условий при открытии модалок, пересмотрел роутинг в сторону использования useHistory в части открытия модалок и обновления страницы, почистил проект от Any типов

gennady-bars commented on package.json:
```diff
@@ -31,6 +32,8 @@
   },
   "scripts": {
+    "predeploy": "npm run build",
+    "deploy": "gh-pages -d build",
```
* Вам нужно добавить скрипт запуска `cypress` тестов ` "cypress": "cypress open cache clear",` в `package.json`

gennady-bars commented on cypress.config.ts:
```diff
@@ -0,0 +1,10 @@
+
+export default defineConfig({
+  e2e: {
+    baseUrl: "http://localhost:3000",
```
* Отлично, что базовый урл вынесен в конфиг

gennady-bars commented on cypress/e2e/constructor.cy.js:
```diff
@@ -0,0 +1,80 @@
+    })
+
+    it("should show modal with ingredient info after click on it", () => {
+        cy.get("[data-testid=ingredient_card]").first().click();
```
`"[data-testid=ingredient_card]"` повторяется 3 раза в тестах

gennady-bars commented on cypress/e2e/constructor.cy.js:
```diff
@@ -0,0 +1,80 @@
+
+    it("should show modal with ingredient info after click on it", () => {
+        cy.get("[data-testid=ingredient_card]").first().click();
+        cy.get("[id=modalRoot]").should("contain.text","Детали ингредиента");
```
`"[id=modalRoot]"` повторяется 7 раз

gennady-bars replied:
Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора.

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная



* Все юнит-тесты (83 шт) проходят успешно
* Все cypress-тесты проходят успешно
* Код тестов аккуратный и понятный
* Отлично, что базовый урл вынесен в конфиг


но есть некоторые недочеты: 

* если вставить ссылку на заказ в браузер, должна появляться уже **страница** с деталями, а не попап. Сейчас при клике сбоку закрывается вся вкладка браузера. Но это должна быть именно **страница**
* Вам нужно добавить скрипт запуска `cypress` тестов ` "cypress": "cypress open cache clear",` в `package.json`
* Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора.


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

anufrp committed:
Исправления после ревью

Исправлено отображение данных заказа на отдельной странице без модалки, добавлен скрипт запуска cypress, повторяющиеся селекторы вынесены в константу в тестах cypress

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.


Удачи в Вашей новой профессии.

**Несколько советов для дальнейшего развития:**

* сверстайте несколько небольших проектов для себя, для закрепления знаний, и чтобы можно добавить их в портфолио
* ходите на собеседования: опыт прохождения собеседований не менее важен знаний теории, и Вы поймете, какие знания требуются на рынке, и где Вы проседаете
* читайте [habr](https://habr.com/ru/) и слушаете подкаксты: они очень хорошо расширяют кругозор и позволяют узнать, что происходит в отрасли:

https://soundcloud.com/everyonecan

https://soundcloud.com/podlodka

https://soundcloud.com/devschacht

https://soundcloud.com/begebot

https://soundcloud.com/web-standards

https://www.youtube.com/channel/UCZeU17nbVfzczAkJVTay9vw
